tests/test_schema.py

```diff
@@ -2766,3 +2767,79 @@ class Meta:
         dumped = OSchema().dump({"foo": 42, "bar": 24})
         assert isinstance(dumped, OrderedDict)
         assert "bar" not in dumped
+
+
+class TestCombineOpts:
+    def test_different_opts(self):
+        """
+        Create two options classes, each belonging to a schema, then combine the two
+        schemas using inheritance. When combine_opts == True, we should have both options
+        being stored in self.opts. Otherwise, we should only get options defined in the
+        first options class stored in self.opts.
+        """
+
+        class Opts1(SchemaOpts):
+            def __init__(self, meta, **kwargs):
+                super().__init__(meta, **kwargs)
+                self.opt_1 = getattr(meta, "opt_1", None)
+
+        class Schema1(Schema):
+            OPTIONS_CLASS = Opts1
+            Field1 = fields.String()
+
+        class Opts2(SchemaOpts):
+            def __init__(self, meta, **kwargs):
+                super().__init__(meta, **kwargs)
+                self.opt_2 = getattr(meta, "opt_2", None)
+
+        class Schema2(Schema):
+            OPTIONS_CLASS = Opts2
+            Field2 = fields.String()
+
+        class NotCombined(Schema1, Schema2, combine_opts=False):
+            class Meta:
+                opt_1 = True
+                opt_2 = True
+
+        class Combined(Schema1, Schema2, combine_opts=True):
+            class Meta:
+                opt_1 = True
+                opt_2 = True
+
+        # Without using combine_opts, we should only get options defined by the first
+        # options class
+        assert NotCombined().opts.opt_1
+        assert not hasattr(NotCombined().opts, "opt_2")
+
+        # However when we use combine_opts=True, we should get options defined by both
+        # options classes
+        assert Combined().opts.opt_1
+        assert Combined().opts.opt_2
+
+    def test_same_opts(self):
+        """
+        Ensure this behaviour still works if the two parent schemas have the same opts
+        class
+        """
+
```

Comment:
I don't get the purpose of this test.

In fact, I don't understand the purpose of the `Meta` class attribute in both tests.

Reply:
The class `Meta` sets the values of fields on the options class. It's just normal Marshmallow behaviour. 

This particular test basically just checks that using `combine_opts=True` works for two schemas with the same options class. Basically just checking backwards compatibility.